modules/telestion-application/src/main/java/de/wuespace/telestion/application/loader/Loader.java

```diff
@@ -0,0 +1,350 @@
+	 * <p>
+	 * <em>Call chain</em>:
+	 * <strong><code>onBeforeVerticleDeploy</code></strong> ➔
+	 * {@link #onAfterVerticleDeploy(Promise, Deployment) onAfterVerticleDeploy}
+	 *
+	 * @param completePromise the promise that indicates the completion state of this loader
+	 * @param deployment      the deployment which contains the {@link io.vertx.core.Verticle Verticle} that deploys
+	 */
+	void onBeforeVerticleDeploy(Promise<Void> completePromise, Deployment deployment) throws Exception;
+
+	/**
+	 * Gets called, after the {@link ConfigDeploymentLauncher Launcher} deploys a new
+	 * {@link io.vertx.core.Verticle Verticle}.
+	 * <p>
+	 * Use this event if you want to run steps after a verticle deploys.
+	 * <p>
+	 * The passed promise should resolve when this loader finishes its execution and fail if the loader encounter
+	 * an error during execution.
+	 * <p>
+	 * The launcher calls this event handler. <strong>Don't call it by yourself.</strong>
+	 * <p>
+	 * <em>Call chain</em>:
+	 * {@link #onBeforeVerticleDeploy(Promise, Deployment) onBeforeVerticleDeploy} ➔
+	 * <strong><code>onAfterVerticleDeploy</code></strong>
+	 *
+	 * @param completePromise the promise that indicates the completion state of this loader
+	 * @param deployment      the deployment which contains the {@link io.vertx.core.Verticle Verticle} that deploys
+	 */
+	void onAfterVerticleDeploy(Promise<Void> completePromise, Deployment deployment) throws Exception;
+
+	/**
+	 * Gets called, before the {@link ConfigDeploymentLauncher Launcher} un-deploys a running
+	 * {@link io.vertx.core.Verticle Verticle}.
+	 * <p>
+	 * Use this event if you want to run steps before a verticle un-deploys.
+	 * <p>
+	 * The passed promise should resolve when this loader finishes its execution and fail if the loader encounter
+	 * an error during execution. A failed promise <strong>prevents</strong> the verticle from un-deploying.
+	 * <p>
+	 * The launcher calls this event handler. <strong>Don't call it by yourself.</strong>
+	 * <p>
+	 * <em>Call chain</em>:
+	 * <strong><code>onBeforeVerticleUndeploy</code></strong> ➔
+	 * {@link #onAfterVerticleUndeploy(Promise, Deployment) onAfterVerticleUndeploy}
+	 *
+	 * @param completePromise the promise that indicates the completion state of this loader
+	 * @param deployment      the deployment which contains the {@link io.vertx.core.Verticle Verticle} that un-deploys
+	 */
+	void onBeforeVerticleUndeploy(Promise<Void> completePromise, Deployment deployment) throws Exception;
+
+	/**
+	 * Gets called, after the {@link ConfigDeploymentLauncher Launcher} un-deploys a running
+	 * {@link io.vertx.core.Verticle Verticle}.
+	 * <p>
+	 * Use this event if you want to run steps after a verticle un-deploys.
+	 * <p>
+	 * The passed promise should resolve when this loader finishes its execution and fail if the loader encounter
+	 * an error during execution. A failed promise <strong>prevents</strong> the verticle from un-deploying.
```

Comment:
Wait a second? This gets called after the un-deployment is already done, right? Does it re-deploy the Verticle to prevent un-deployment? 😉

Reply:
Oops. You've catched a copy-paste error.

Reply:
Done in [`46e0777` (#501)](https://github.com/wuespace/telestion-core/pull/501/commits/46e07771cc2222f313d84472f69aa37a80b05033). Please re-review if you don't like it. :wink: